lib/utils/forkedblockchain.js

```diff
@@ -41,6 +41,12 @@ function ForkedBlockchain(options) {
 
   BlockchainDouble.call(this, options);
 
+  this.createVMFromStateTrie = function() {
+    var vm = BlockchainDouble.prototype.createVMFromStateTrie.apply(this, arguments);
+    this.patchVM(vm);
```

Comment:
Forking requires that the vm be "special" otherwise things get all broken on us. When forking calls `processCall` over in `BlockChainDouble` (or traces a transaction, which we actually don't test for) it needs to have some special monkey patching set up deep in the VM's stateManager in order to continue working.